File: UPCRideAndroid/app/src/main/java/com/example/upcride/activities/SolicitudesFragment.java
package com.example.upcride.activities;


import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.upcride.R;
import com.example.upcride.model.Solicitud;

import java.util.ArrayList;
import java.util.List;

/**
 * A simple {@link Fragment} subclass.
 */
public class SolicitudesFragment extends Fragment {

    private RecyclerView recyclerViewSolicitud;
    private AdaptadorSolicitudes adaptadorSolicitudes;

    public SolicitudesFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View vista = inflater.inflate(R.layout.fragment_solicitudes, container, false);

        recyclerViewSolicitud = (RecyclerView) vista.findViewById(R.id.recycle_solis);
        recyclerViewSolicitud.setLayoutManager(new LinearLayoutManager(getContext()));

        adaptadorSolicitudes = new AdaptadorSolicitudes(carteraSolicitudes());
        recyclerViewSolicitud.setAdapter(adaptadorSolicitudes);

        return vista;
    }

    public List<SolicitudModelo> carteraSolicitudes(){
        List<SolicitudModelo> solicitudes = new ArrayList<>();
        solicitudes.add(new SolicitudModelo("Brian", "09:10", "Solicitó un viaje", R.drawable.ricardo));
        solicitudes.add(new SolicitudModelo("Bonifacio", "22:10", "Realizó una reseña", R.drawable.queso));
        solicitudes.add(new SolicitudModelo("Emilio", "21:10", "Solicitó un viaje", R.drawable.calamardo));
        solicitudes.add(new SolicitudModelo("Juanelv", "09:10", "Solicitó un viaje", R.drawable.ricardo));
        return solicitudes;
    }

}

File: UPCRideKotlin/app/src/main/java/com/example/upcridekotlin/activities/SolicitudesFragment.kt
package com.example.upcridekotlin.activities


import android.os.Bundle

import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

import com.example.upcridekotlin.R
import com.example.upcridekotlin.interfaces.SolicitudApiService
import com.example.upcridekotlin.interfaces.ViajeApiService
import com.example.upcridekotlin.model.Solicitud
import com.example.upcridekotlin.model.SolicitudModelo
import kotlinx.android.synthetic.main.item_viaje_historial.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


import java.util.ArrayList

/**
 * A simple [Fragment] subclass.
 */
class SolicitudesFragment : Fragment() {

    private var recyclerViewSolicitud: RecyclerView? = null
    private var adaptadorSolicitudes: AdaptadorSolicitudes? = null
    private var solicitudService: SolicitudApiService? = null
    private var conductorId: Int? = 0



    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val vista = inflater.inflate(R.layout.fragment_solicitudes, container, false)

        recyclerViewSolicitud = vista.findViewById(R.id.recycle_solis)
        recyclerViewSolicitud!!.layoutManager = LinearLayoutManager(context)

        adaptadorSolicitudes = AdaptadorSolicitudes(carteraSolicitudes())
        recyclerViewSolicitud!!.adapter = adaptadorSolicitudes



        return vista
    }




    fun carteraSolicitudes(): List<SolicitudModelo> {
        val solicitudes = ArrayList<SolicitudModelo>()
        solicitudes.add(
            SolicitudModelo(
                "Burga",
                "09:10",
                "Baja en 3 calles",
                R.drawable.user
            )
        )
        solicitudes.add(
            SolicitudModelo(
                "Semanche",
                "22:10",
                "Porfa tengo un bautizo",
                R.drawable.user
            )
        )
        solicitudes.add(
            SolicitudModelo(
                "Emilio",
                "21:10",
                "Muchachos me canse",
                R.drawable.user
            )
        )
        solicitudes.add(
            SolicitudModelo(
                "Elba",
                "09:10",
                "Tu, yo  y un codelab, piensalo",
                R.drawable.user
            )
        )
        return solicitudes
    }

}// Required empty public constructor
